package.json

```diff
@@ -86,9 +88,11 @@
     "enzyme": "^2.2.0",
     "eslint": "^2.9.0",
     "eslint-plugin-babel": "^3.2.0",
+    "eslint-plugin-flow-vars": "^0.4.0",
     "eslint-plugin-material-ui": "./packages/eslint-plugin-material-ui",
     "eslint-plugin-mocha": "^3.0.0",
     "eslint-plugin-react": "^5.0.1",
+    "flow-bin": "^0.27.0",
```

Comment:
Facebook has recently releases `0.28.0` with a new union types implementation.
Have you tried upgrading the dependency?